server/src/main/java/com/hedera/block/server/persistence/storage/write/BlockAsDirWriter.java

```diff
@@ -169,7 +171,12 @@ private void resetState(@NonNull final BlockItem blockItem) throws IOException {
         repairPermissions(blockNodeRootPath);
 
         // Construct the path to the block directory
-        FileUtils.createPathIfNotExists(calculateBlockPath(), DEBUG, filePerms);
+        createPathIfNotExists(
+                calculateBlockPath(),
+                DEBUG,
+                filePerms,
+                BLOCK_NODE_ROOT_DIRECTORY_SEMANTIC_NAME,
+                true);
```

Comment:
Just use the override with defaults for this class; it's not a production class.
In the production classes we'll _always_ set file and folder permissions, which should _not_ have a default outside of the configuration records.

Reply:
Agreed about that, but since this class is not a production class, do you envision that we will ever need anything else than the defaults, meaning to use the overloaded method? If not, can we then consider to remove the `filePerms` field, as well as in the builder for this class and also to do the same in the `BlockAsDirReader` and it's builder, since it is the same thing there.

Reply:
The point of the overload is to enable a class like this to accept defaults without having to manage that default value.

It's fine to remove the class attribute, but it also makes sense to _allow_ for override in a test/development class like this.  If the value is not set (it is `null`) then it was not overridden and the code will call the overload without a permissions.  If the value _is_ set (it is not `null`), then the code calls the version that requires permissions.  The tricky part is that file and folder _must not_ have the same permissions.
For now, just removing the permissions support in this class and calling the overload without the permissions parameters is acceptable.

Reply:
✅ Done. Keep in mind that on a second inspection, the perms were used not only for the create path utility, but also for other reasons both in `BlockAsDirWriter` and `BlockAsDirReader`. That being said, if I left the field `null` it would have been quite a pain again, since I do need to provide permissions for the other usages of the `filePerms` field OR I must have not executed the logic that needed the non-null `filePerms`. I do not think the latter must happen, so instead of the possibility of the field to be `null`, I provide the field sane defaults (local to the respective class in order to fit the philosophy) in case the `filePerms` argument of the constructor is `null`.

Reply:
file permissions is one of those things that defaults are a bit risky.
I still recommend changing the code to handle a null value (and not have a local default value); that can be done separately, however.

Reply:
I am seeing a number of possible errors as I look at the full _usage_ of this permissions variable.  It seems everything in this class is assuming identical permissions for both files and folders.
This class probably needs some added overhaul to handle all this correctly, so perhaps a separate issue is appropriate?

Additional Discussion:
This is _exactly_ why having a single permissions value is a problem.
I recommend, if we continue to set permissions at all (see below), that _this class_ split file and folder permissions to make them expressly different, and ensure that files _do not_ have the execute bit set at any point.  I also recommend that group and other permissions be limited to, at most, read for files and read+execute for folders.

Ideally just _not setting permissions_ for files is better. The server should be relying on system umask (set in the launch script for the server) to correctly set permissions for new files and folders (something like umask 033 would be appropriate). This would place system file permissions _properly_ in the control of system operations.

Reply:
@jsync-swirlds I opened an issue #319 for this thread of comments here.